world-chain-builder/crates/world/pool/src/tx.rs

```diff
@@ -14,31 +14,31 @@ use reth_primitives::{kzg::KzgSettings, Recovered};
 use reth_primitives_traits::InMemorySize;
 use revm_primitives::{Address, TxKind, B256, U256};
 use thiserror::Error;
-use world_chain_builder_pbh::payload::PBHValidationError;
+use world_chain_builder_pbh::payload::{PBHPayload, PBHValidationError};
 
 #[derive(Debug, Clone)]
 pub struct WorldChainPooledTransaction {
     pub inner: OpPooledTransaction,
-    pub valid_pbh: bool,
+    pub payload: Option<Vec<PBHPayload>>,
 }
 
 pub trait WorldChainPoolTransaction: EthPoolTransaction {
-    fn valid_pbh(&self) -> bool;
-    fn set_valid_pbh(&mut self);
+    fn set_pbh_payloads(&mut self, payload: Vec<PBHPayload>);
     fn conditional_options(&self) -> Option<&TransactionConditional>;
+    fn pbh_payload(&self) -> Option<&Vec<PBHPayload>>;
 }
 
```

Comment:
Consider updating to use a slice `fn pbh_payload(&self) -> Option<&[PBHPayload]>;`